In [1]:
%matplotlib inline
import matplotlib.pyplot as plt
import numpy as np, pandas as pd
# pd.set_option('display.max_rows', 8)

import gbd_mapping as gm
import vivarium_inputs.interface as vii

!date
!whoami

Tue Dec 17 18:29:57 PST 2019
ndbs


In [12]:
import vivarium_inputs.utility_data as viu

In [2]:
vii.get_age_bins()

Empty DataFrame
Columns: []
Index: [([0.0, 0.01917808), Early Neonatal), ([0.01917808, 0.07671233), Late Neonatal), ([0.07671233, 1.0), Post Neonatal), ([1.0, 5.0), 1 to 4), ([5.0, 10.0), 5 to 9), ([10.0, 15.0), 10 to 14), ([15.0, 20.0), 15 to 19), ([20.0, 25.0), 20 to 24), ([25.0, 30.0), 25 to 29), ([30.0, 35.0), 30 to 34), ([35.0, 40.0), 35 to 39), ([40.0, 45.0), 40 to 44), ([45.0, 50.0), 45 to 49), ([50.0, 55.0), 50 to 54), ([55.0, 60.0), 55 to 59), ([60.0, 65.0), 60 to 64), ([65.0, 70.0), 65 to 69), ([70.0, 75.0), 70 to 74), ([75.0, 80.0), 75 to 79), ([80.0, 85.0), 80 to 84), ([85.0, 90.0), 85 to 89), ([90.0, 95.0), 90 to 94), ([95.0, 125.0), 95 plus)]

In [3]:
age_bins = vii.get_age_bins()
age_bins.index

MultiIndex(levels=[[[0.0, 0.01917808), [0.01917808, 0.07671233), [0.07671233, 1.0), [1.0, 5.0), [5.0, 10.0), [10.0, 15.0), [15.0, 20.0), [20.0, 25.0), [25.0, 30.0), [30.0, 35.0), [35.0, 40.0), [40.0, 45.0), [45.0, 50.0), [50.0, 55.0), [55.0, 60.0), [60.0, 65.0), [65.0, 70.0), [70.0, 75.0), [75.0, 80.0), [80.0, 85.0), [85.0, 90.0), [90.0, 95.0), [95.0, 125.0)], ['1 to 4', '10 to 14', '15 to 19', '20 to 24', '25 to 29', '30 to 34', '35 to 39', '40 to 44', '45 to 49', '5 to 9', '50 to 54', '55 to 59', '60 to 64', '65 to 69', '70 to 74', '75 to 79', '80 to 84', '85 to 89', '90 to 94', '95 plus', 'Early Neonatal', 'Late Neonatal', 'Post Neonatal']],
           codes=[[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22], [20, 21, 22, 0, 9, 1, 2, 3, 4, 5, 6, 7, 8, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]],
           names=['age', 'age_group_name'])

In [6]:
age_bins.index.get_level_values('age')

IntervalIndex([[0.0, 0.01917808), [0.01917808, 0.07671233), [0.07671233, 1.0), [1.0, 5.0), [5.0, 10.0) ... [75.0, 80.0), [80.0, 85.0), [85.0, 90.0), [90.0, 95.0), [95.0, 125.0)],
              closed='left',
              name='age',
              dtype='interval[float64]')

### Ok, so the dataframe has an interval index in its first level... what am I supposed to index with it?

In [7]:
age_bins.reset_index()

age  age_group_name
0          [0.0, 0.01917808)  Early Neonatal
1   [0.01917808, 0.07671233)   Late Neonatal
2          [0.07671233, 1.0)   Post Neonatal
3                 [1.0, 5.0)          1 to 4
4                [5.0, 10.0)          5 to 9
5               [10.0, 15.0)        10 to 14
6               [15.0, 20.0)        15 to 19
7               [20.0, 25.0)        20 to 24
8               [25.0, 30.0)        25 to 29
9               [30.0, 35.0)        30 to 34
10              [35.0, 40.0)        35 to 39
11              [40.0, 45.0)        40 to 44
12              [45.0, 50.0)        45 to 49
13              [50.0, 55.0)        50 to 54
14              [55.0, 60.0)        55 to 59
15              [60.0, 65.0)        60 to 64
16              [65.0, 70.0)        65 to 69
17              [70.0, 75.0)        70 to 74
18              [75.0, 80.0)        75 to 79
19              [80.0, 85.0)        80 to 84
20              [85.0, 90.0)        85 to 89
21              [90.0, 95.0)        90 to 94
22             [95.0, 125.0)         95 plus

In [8]:
type(age_bins.reset_index().loc[0,'age'])

pandas._libs.interval.Interval

In [9]:
age_bins.index.get_level_values('age_group_name')

Index(['Early Neonatal', 'Late Neonatal', 'Post Neonatal', '1 to 4', '5 to 9',
       '10 to 14', '15 to 19', '20 to 24', '25 to 29', '30 to 34', '35 to 39',
       '40 to 44', '45 to 49', '50 to 54', '55 to 59', '60 to 64', '65 to 69',
       '70 to 74', '75 to 79', '80 to 84', '85 to 89', '90 to 94', '95 plus'],
      dtype='object', name='age_group_name')

### Well, at least the names are just strings

In [10]:
type(age_bins)

pandas.core.frame.DataFrame

In [11]:
age_bins.columns

Index([], dtype='object')

### So the dataframe is empty -- it just consists of an index... um, why?

### If we pull age bins from `vivarium_inputs.utility_data` instead, we can get the age group ids:

In [13]:
viu.get_age_bins()

age_group_id  age_group_name  age_start     age_end
0              2  Early Neonatal   0.000000    0.019178
1              3   Late Neonatal   0.019178    0.076712
2              4   Post Neonatal   0.076712    1.000000
3              5          1 to 4   1.000000    5.000000
4              6          5 to 9   5.000000   10.000000
5              7        10 to 14  10.000000   15.000000
6              8        15 to 19  15.000000   20.000000
7              9        20 to 24  20.000000   25.000000
8             10        25 to 29  25.000000   30.000000
9             11        30 to 34  30.000000   35.000000
10            12        35 to 39  35.000000   40.000000
11            13        40 to 44  40.000000   45.000000
12            14        45 to 49  45.000000   50.000000
13            15        50 to 54  50.000000   55.000000
14            16        55 to 59  55.000000   60.000000
15            17        60 to 64  60.000000   65.000000
16            18        65 to 69  65.000000   70.000000
17            19        70 to 74  70.000000   75.000000
18            20        75 to 79  75.000000   80.000000
19            30        80 to 84  80.000000   85.000000
20            31        85 to 89  85.000000   90.000000
21            32        90 to 94  90.000000   95.000000
22           235         95 plus  95.000000  125.000000

In [16]:
viu.get_location_id('Nigeria')

214

In [23]:
gm.causes.neural_tube_defects.restrictions

Restrictions(
male_only=False,
female_only=False,
yll_only=False,
yld_only=False,
yll_age_group_id_start=2,
yll_age_group_id_end=17,
yld_age_group_id_start=2,
yld_age_group_id_end=235,
violated=[yll_age_restriction_violated_by_deaths])

### Age groups for restrictions:

2: Early Neonatal

17: 60 - 64

235: 95+

## Ok, let's try pulling some data for neural tube defects

In [18]:
ntd_birth_prev_nig = vii.get_measure(gm.causes.neural_tube_defects, 'birth_prevalence', 'Nigeria')
ntd_birth_prev_nig.head()

________________________________________________________________________________
[Memory] Calling vivarium_gbd_access.gbd.get_birth_prevalence...
get_birth_prevalence(entity_id=cid(642), location_id=214, entity_type='cause')
____________________________________________get_birth_prevalence - 47.8s, 0.8min


draw_0    draw_1    draw_2    draw_3  \
location sex    year                                                   
Nigeria  Female [1990, 1991)  0.002767  0.002659  0.002663  0.002957   
                [1991, 1992)  0.002704  0.002660  0.002661  0.002859   
                [1992, 1993)  0.002638  0.002659  0.002654  0.002759   
                [1993, 1994)  0.002570  0.002654  0.002644  0.002658   
                [1994, 1995)  0.002502  0.002648  0.002631  0.002558   

                                draw_4    draw_5    draw_6    draw_7  \
location sex    year                                                   
Nigeria  Female [1990, 1991)  0.002778  0.002998  0.002927  0.002765   
                [1991, 1992)  0.002755  0.002972  0.002919  0.002760   
                [1992, 1993)  0.002730  0.002946  0.002906  0.002752   
                [1993, 1994)  0.002702  0.002918  0.002891  0.002743   
                [1994, 1995)  0.002671  0.002889  0.002872  0.002733   

                                draw_8    draw_9  ...  draw_990  draw_991  \
location sex    year                              ...                       
Nigeria  Female [1990, 1991)  0.002567  0.002761  ...  0.002498  0.002496   
                [1991, 1992)  0.002527  0.002712  ...  0.002496  0.002497   
                [1992, 1993)  0.002484  0.002661  ...  0.002488  0.002495   
                [1993, 1994)  0.002440  0.002607  ...  0.002476  0.002490   
                [1994, 1995)  0.002394  0.002552  ...  0.002461  0.002483   

                              draw_992  draw_993  draw_994  draw_995  \
location sex    year                                                   
Nigeria  Female [1990, 1991)  0.002691  0.003098  0.002718  0.002865   
                [1991, 1992)  0.002685  0.003072  0.002728  0.002821   
                [1992, 1993)  0.002674  0.003044  0.002734  0.002771   
                [1993, 1994)  0.002660  0.003015  0.002737  0.002718   
                [1994, 1995)  0.002643  0.002985  0.002738  0.002663   

                              draw_996  draw_997  draw_998  draw_999  
location sex    year                                                  
Nigeria  Female [1990, 1991)  0.002722  0.003032  0.002873  0.002268  
                [1991, 1992)  0.002717  0.002953  0.002828  0.002292  
                [1992, 1993)  0.002709  0.002874  0.002782  0.002309  
                [1993, 1994)  0.002699  0.002797  0.002736  0.002320  
                [1994, 1995)  0.002687  0.002721  0.002689  0.002328  

[5 rows x 1000 columns]

In [19]:
# 1 age group x 2 sexes x 28 yeears = 56 rows
ntd_birth_prev_nig.shape

(56, 1000)

### Try pulling raw data to compare results

From GBD id tables (https://shiny.ihme.washington.edu/content/88/):

age_group_id=164: "birth"

measure_id=6: "incidence"

metric_id=3: "rate"

sex_id=1: "male", sex_id=2: "female"

In [20]:
ntd_raw_birth_prev_nig = vii.get_raw_data(gm.causes.neural_tube_defects, 'birth_prevalence', 'Nigeria')
ntd_raw_birth_prev_nig.head()

________________________________________________________________________________
[Memory] Calling vivarium_gbd_access.gbd.get_birth_prevalence...
get_birth_prevalence(entity_id=cid(642), location_id=214, entity_type='cause')
____________________________________________get_birth_prevalence - 25.8s, 0.4min


age_group_id  cause_id    draw_0    draw_1   draw_10  draw_100  draw_101  \
0           164       642  0.002844  0.003004  0.002790  0.002968  0.002484   
1           164       642  0.002767  0.002659  0.002569  0.002788  0.002673   
2           164       642  0.002815  0.002965  0.002771  0.002949  0.002478   
3           164       642  0.002704  0.002660  0.002550  0.002793  0.002607   
4           164       642  0.002787  0.002923  0.002751  0.002929  0.002472   

   draw_102  draw_103  draw_104  ...  draw_995  draw_996  draw_997  draw_998  \
0  0.003400  0.002638  0.002604  ...  0.002839  0.002888  0.002877  0.002726   
1  0.003183  0.002643  0.002767  ...  0.002865  0.002722  0.003032  0.002873   
2  0.003355  0.002643  0.002626  ...  0.002866  0.002865  0.002910  0.002760   
3  0.003143  0.002620  0.002738  ...  0.002821  0.002717  0.002953  0.002828   
4  0.003301  0.002647  0.002645  ...  0.002892  0.002841  0.002942  0.002795   

   draw_999  location_id  measure_id  sex_id  year_id  metric_id  
0  0.002477          214           6       1     1990          3  
1  0.002268          214           6       2     1990          3  
2  0.002450          214           6       1     1991          3  
3  0.002292          214           6       2     1991          3  
4  0.002422          214           6       1     1992          3  

[5 rows x 1007 columns]

In [21]:
ntd_raw_birth_prev_nig.shape

(56, 1007)

### It looks like the numbers match if you match up the rows correctly

E.g. row 1 from `get_measure` looks like row 2 from `get_raw_data`.